test/functional/api/server_requests_spec.lua

```diff
@@ -304,14 +304,30 @@ describe('server -> client', function()
       connect_test(server, 'pipe', address)
     end)
 
-    it('via ip address', function()
+    it('via ipv4 address', function()
       local server = spawn(nvim_argv)
       set_session(server)
       local address = funcs.serverstart("127.0.0.1:")
+      if #address == 0 then
```

Comment:
If serverstart() returns empty for some other reason, this won't catch the problem.

But checking for ipv6 support via the shell is annoying and very platform-dependent. :(